Enable image editing on private Atomic sites

jsnajdr opened the pull request:
Since the "private by default" project for Atomic sites, we have an `atomic-proxy` REST endpoint to download media files that are protected by authentication. It's used to display media file thumbnails in the Media Gallery.

This PR uses the same `atomic-proxy` REST endpoint for editing images. As the `atomic-proxy` requests are regular requests to `public-api.wordpress.com`, they don't need any CORS headers and can be easily loaded into canvas.

Image editing on private Simple sites remains disabled because there we need to download the media files directly, and we'll need to add the right `Access-Control-Allow-Origin` and `Access-Control-Allow-Credentials` headers to server responses to make them work with a canvas.

Besides the core work enabling `atomic-proxy`, there are a few independent related commits:
- remove the `post-editor/image-editor` feature flag that's always `true`
- remove legacy Jetpack version check (`> 4.7-alpha`). @tyxla enjoys removing these 🙂 
- in the original non-`atomic-proxy` case, use Fetch API instead of XHR to fetch the image


**How to test:**
- go to an Atomic site that is Private
- open its Media section in Calypso
- edit an image, make sure that the detail modal has an "Edit Image" button
- verify that after clicking that button, you can rotate/crop the image and save it back

jsnajdr committed:
Remove post-editor/image-editor feature flag

jsnajdr committed:
Remove legacy Jetpack version check

jsnajdr committed:
Fetch image blob with fetch API

jsnajdr committed:
Fetch images for Atomic Private through proxy

adamziel commented:
Just eyeballing 👀  the PR for now it looks pretty good and the E2E test failures seem unrelated, thank you for working on this  @jsnajdr!

adamziel commented:
I just tried to test this one without setting up my local env, but it doesn't seem like the media proxy works with `calypso.live`:

`{"body":{"code":"rest_forbidden","message":"Sorry, you are not allowed to do that.","data":{"status":403}},"status":403,"headers":[]}`

<img width="1708" alt="Zrzut ekranu 2021-03-25 o 10 49 53" src="https://user-images.githubusercontent.com/205419/112453524-14c49c80-8d58-11eb-9b44-47950e57eb91.png">

For comparison, it works just fine on wp.com:

<img width="1713" alt="Zrzut ekranu 2021-03-25 o 10 49 45" src="https://user-images.githubusercontent.com/205419/112453549-1b531400-8d58-11eb-8aaf-3fd934866eed.png">

**Edit:** I see you already addressed that in D58521-code

adamziel reviewed (approved):
I worked perfectly in my testing, thank you! 👍 The only thing I would add is a loading indicator as large images sometimes take 5+ seconds to load on my end and I only see gray background until they do:

<img width="1825" alt="Zrzut ekranu 2021-03-25 o 11 04 43" src="https://user-images.githubusercontent.com/205419/112455685-3e7ec300-8d5a-11eb-8fa1-e0d751ff2da3.png">

jsnajdr committed:
DetailPreviewImage: set loading initially to true

jsnajdr committed:
ProxiedImage: load updated image on props change

jsnajdr commented:
> it doesn't seem like the media proxy works with `calypso.live`:

This is just an omission in the server-side origin check (`Abstract_Proxy::is_requested_by_trusted_client`) that allows `http://calypso.localhost:3000` and all other common Calypso origins, but not `*.calypso.live`. Can be fixed by a simple wpcom patch.

> The only thing I would add is a loading indicator as large images sometimes take 5+ seconds to load

Turns out that this is just a little bug in the the `DetailPreviewImage` component that doesn't set its `loading` state initially to `true`. Fixed.

I also fixed another little bug in `ProxiedImage` that didn't load a new image when its `filePath` et al. props change. It was observable when selecting two or more images and then rotating with left and right arrows between them.
